### PR TITLE
Replace deprecated python 3.8 code and add missing __init__.py files

### PR DESCRIPTION
**Pull request by ksuszka**
#### Description

- Add missing `srunner/osc2/__init__.py` file and fix a misspell in `srunner/osc2/ast_manager/__init__.py` file name.
- Replace xml.etree.ElementTree.Element.getchildren method which is deprecated from python 3.2 and removed in python 3.9 with a suggested replacement https://docs.python.org/3.8/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren

#### Where has this been tested?

  * **Platform(s):** Ubuntu 22.04
  * **Python version(s):** 3.10.12
  * **Unreal Engine version(s):** 4.26
  * **CARLA version:** 0.9.15

#### Possible Drawbacks

None.

- - -
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/scenario_runner/1053)

### CODE REVIEW

**Review by ttgamage — APPROVED** [review]

**Comment by ksuszka** [comment]
@ttgamage I'm not sure what merge policy is here in place. When I clicked notification email about your approval, there was a button here to rebase this change on top of the latest changes so I clicked it, but it seems that you need to approve it again.

**Review by Daraan — COMMENTED** [review]
This PR definitely will be needed for later python version and for those it can be approved.

Some things on my mind: 
 - Are we sure that it also works with the current lower versions for backward compatibility, i.e. do these changes work with the current requirements.txt?
- The requirements.txt might need to be updated accordingly, or a second introduced for the later python versions.

File: srunner/osc2/__init__.py
[the remark this answers is not in the document]
Comment (ksuszka) [reply srunner/osc2/__init__.py:1]: The getchildren method [was deprecated in python 2.7](https://docs.python.org/2.7/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.getchildren) released in 2010., 8 years before this repository was created. So it should be safe to drop it.

As for `__init__.py` files. Although Python doesn't require presence of this file anymore it is still required by some libraries, i.e. setuptools. I found out about this the hard way when I tried to pack this repository into a python package for an internal distribution.

**Review by Daraan — COMMENTED** [review]
:lgtm: I tested with the requirements file and Python3.7

`list(element)` also works in Python2 (which is not really compatible with the requirements given)

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/scenario_runner/1053#-:-NvaSH107TftgqHIF5Ox:b-li7tbq)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ksuszka)

**Review by Daraan — APPROVED** [review]

**Review by glopezdiest — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/scenario_runner/1053)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ksuszka)

**Comment by glopezdiest** [comment]
Nothign to add to the discussion, approved

**Pull request merged**